src/DynamoCoreWpf/UI/Prompts/DynamoMessageBox.xaml.cs

```diff
@@ -168,56 +201,51 @@ public static MessageBoxResult Show(string messageBoxText, string caption)
         /// Shows or hides buttons on the view as necessary.
         /// </summary>
         /// <param name="messageBoxButton"></param>
-        private void ConfigureButtons(MessageBoxButton messageBoxButton)
+        /// /// <param name="buttonNames">names that will be used to override the standard button names.
+        /// Number of names must match the number of visible buttons set by messageBoxButton parameter.</param>
+        internal void ConfigureButtons(MessageBoxButton messageBoxButton, IEnumerable<string> buttonNames = null)
         {
             switch (messageBoxButton)
             {
                 case MessageBoxButton.OK:
+                    if(buttonNames!=null && buttonNames.Count() == 1)
+                    {
+                        OkButton.Content = buttonNames.ElementAt(0);
+                    }
                     OkButton.Visibility = Visibility.Visible;
                     break;
                 case MessageBoxButton.OKCancel:
+                    if (buttonNames != null && buttonNames.Count() == 2)
+                    {
+                        OkButton.Content = buttonNames.ElementAt(0);
+                        CancelButton.Content = buttonNames.ElementAt(1);
+                    }
```

Comment:
What if buttonNames count is 1 here, meaning if there is only one custom name, and the other is the default name?

Reply:
Should we also validate that the number of custom names is equal to the number of buttons and if not, just use the standard names?

Reply:
> Should we also validate that the number of custom names is equal to the number of buttons and if not, just use the standard names?


That is what this does currently, at least thats what I thought I wrote 😉  - if the `buttonNames` count does not equal the expected number of buttons, the content is not modified - (it's set to the original names in the xaml)

Reply:
Ah, sorry, I missed that.